### PR TITLE
Upgrade Yeoman

### PR DESCRIPTION
**Pull request by tomhalex**
This was complaining of vulnerabilities when I tried to `link` it, so I upgraded Yeoman to fix. Had to make some tweaks to index.js, but it appears fine.

Please double-check me on that though - I did some Googling for Yeoman, but that was about all I had

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -16,10 +16,10 @@ function camelcase(name) {
     return words.join("");
 }
 
-module.exports = yeoman.generators.Base.extend({
-    prompting: function () {
-        var done = this.async();
```
Comment (tomhalex) [inline app/index.js:21]: Don't use this anymore - it crashes out at the end of the first method if you do - use the new `await` keyword

File: app/index.js [hunk 2]
```diff
@@ -48,115 +48,111 @@ module.exports = yeoman.generators.Base.extend({
...
-            props.className = camelcase(props.name);
-            props.slug = decompose(props.name).join("-");
 
-            this.props = props;
-            // To access props later use this.props.someOption;
-
-            done();
-        }.bind(this));
-    },
+        const answers = await this.prompt(prompts)
```
Comment (tomhalex) [inline app/index.js:52]: This is the async call

File: app/index.js [hunk 3]
```diff
@@ -48,115 +48,111 @@ module.exports = yeoman.generators.Base.extend({
...
-            for (var i = 0; i < files.length; i++) {
-                this.fs.copyTpl(
-                    this.templatePath(files[i]),
-                    this.destinationPath(files[i]),
-                    this.props
-                );
-            }
-            this.fs.copy(
-                this.templatePath("gradle/wrapper/gradle-wrapper.jar"),
-                this.destinationPath("gradle/wrapper/gradle-wrapper.jar")
```
Comment (tomhalex) [inline app/index.js:84]: We now have to specify all the methods on the base class, NOT as a separate object

File: package.json [hunk 4]
```diff
@@ -26,7 +26,7 @@
   ],
   "dependencies": {
     "chalk": "^1.1.3",
-    "yeoman-generator": "^0.20.3",
+    "yeoman-generator": "^4.2.0",
```
Comment (tomhalex) [inline package.json:29]: Bump Yeoman

**Review by doches — APPROVED** [review]
👍 

Took me embarrassingly long to realise I needed to `npm install` to pick up the newer version of `yeoman-generator` before I could get this to work for me.

**Pull request merged**